src/messaging/ExchangeContext.cpp

```diff
@@ -392,29 +393,52 @@ CHIP_ERROR ExchangeContext::HandleMessage(const PacketHeader & packetHeader, con
     bool isStandaloneAck = payloadHeader.HasMessageType(Protocols::SecureChannel::MsgType::StandaloneAck);
     bool isDuplicate     = msgFlags.Has(MessageFlagValues::kDuplicateMessage);
 
-    CHIP_ERROR err = mDispatch->OnMessageReceived(payloadHeader, packetHeader.GetMessageId(), peerAddress, msgFlags,
-                                                  GetReliableMessageContext());
-    SuccessOrExit(err);
+    auto deffered = MakeDefer([&]() {
+        // Also don't close if there's an outer HandleMessage invocation.  It'll deal with the closing.
```

Comment:
The comment here should be the "The alreadyHandlingMessage check is effectively a workaround" bit that's further below; otherwise this comment makes no sense and neither does the other one; this looks like a merge problem when the `alreadyHandlingMessage` bit got moved around but the comments did not follow it?